src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/FeatureFlagService.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.Internal.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.Razor;
+
+[Export(typeof(IFeatureFlagService))]
+[method: ImportingConstructor]
+internal sealed class FeatureFlagService(
+    [Import(typeof(SAsyncServiceProvider))] IAsyncServiceProvider serviceProvider,
+    JoinableTaskContext joinableTaskContext)
+    : IFeatureFlagService
+{
+    private readonly JoinableTask<IVsFeatureFlags> _getVsFeatureFlagsTask = joinableTaskContext.Factory.RunAsync(
+        () => GetVsFeatureFlagsAsync(serviceProvider));
+
+    private static async Task<IVsFeatureFlags> GetVsFeatureFlagsAsync(IAsyncServiceProvider serviceProvider)
+    {
+        var featureFlags = await serviceProvider.GetFreeThreadedServiceAsync<SVsFeatureFlags, IVsFeatureFlags>().ConfigureAwait(false);
+        Assumes.Present(featureFlags);
+
+        return featureFlags;
+    }
+
+    public bool IsFeatureEnabled(string featureName, bool defaultValue = false)
```

Comment:
Still reviewing, so it may be answered later: 

Should this be an async ValueTask? Wouldn't that ensure we don't have a race condition?

Reply:
I considered that, but it introduces a lot of complexity in the locations where we request `IsFeatureEnabled(...)`, such as the `VisualStudioLanguageServiceFeatureOptions`. However, maybe you're right that I should just do the extra work to make this and `ISettingsPersistenceService` async. It seems to me that it would be too easy to for somebody to import `IFeatureFlagService` in a constructor and call `IsFeatureEnabled(...)` immediately, which would trigger the race condition. I'll go ahead and rework this.

Reply:
Yea, it's unfortunate. A synch version might could be kept around with at least a throw? Constructor could use async to hook up as needed. I'm a bit too fuzzy brain right now to provide more clever recommendations

Reply:
Yeah, I might just throw in the towel and use `AsyncLazy<T>` in `VIsualStudioLanguageServiceFeatureOptions`. :smile: